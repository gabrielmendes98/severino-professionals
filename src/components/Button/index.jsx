import PropTypes from 'prop-types';
import MuiButton from '@material-ui/core/Button';

const Button = ({ children, ...props }) => (
  <MuiButton {...props}>{children}</MuiButton>
);

Button.propTypes = {
  children: PropTypes.any.isRequired,
};

export default Button;
