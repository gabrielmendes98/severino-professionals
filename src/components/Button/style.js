import styled, { css } from 'styled-components';
import MuiButton from '@material-ui/core/Button';
import { marginHandler } from 'commons/styles/margin';
import {
  getContrastColor,
  getColor,
  getDarkerColor,
} from 'commons/styles/util';
import { notForward } from 'commons/styles/config';

export const StyledButton = styled(MuiButton).withConfig(
  notForward('customColor'),
)`
  ${marginHandler};

  ${({ customColor }) =>
    customColor &&
    css`
      background-color: ${({ theme, customColor }) =>
        getColor(theme, customColor)};

      color: ${({ theme, customColor }) =>
        getContrastColor(theme, customColor)};

      &:hover {
        background-color: ${({ theme, customColor }) =>
          getDarkerColor(theme, customColor)};
    `}
`;
