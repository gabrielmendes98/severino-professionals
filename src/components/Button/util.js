export const isDefaultColor = color =>
  ['default', 'inherit', 'primary', 'secondary'].includes(color);
