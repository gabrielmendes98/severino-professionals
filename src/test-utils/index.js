/* eslint-disable react/prop-types */
import { render, configure, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Form, Formik } from 'formik';
import MainProvider from 'commons/providers/MainProvider';

configure({ testIdAttribute: 'id' });

const Wrapper = ({ children }) => (
  <>
    <MainProvider>{children}</MainProvider>
    <div id="modal"></div>
  </>
);
const RouteWrapper = ({ children }) => (
  <BrowserRouter>
    <Wrapper>{children}</Wrapper>
  </BrowserRouter>
);
const FormikWrapper = ({ children, initialValues, ...config }) => (
  <Formik initialValues={initialValues} {...config}>
    <Form>{children}</Form>
  </Formik>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: Wrapper, ...options });

const renderWithRouter = (ui, { route = '/', ...options } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: RouteWrapper, ...options });
};

const asFragment = ui => {
  const { asFragment } = customRender(ui);
  return asFragment(ui);
};

const toMatchSnapshot = (Component, { useRouter = false } = {}) => {
  const wrapper = asFragment(
    useRouter ? <BrowserRouter>{Component}</BrowserRouter> : Component,
  );
  expect(wrapper).toMatchSnapshot();
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export {
  customRender as render,
  screen,
  toMatchSnapshot,
  asFragment,
  renderWithRouter,
  FormikWrapper,
};
