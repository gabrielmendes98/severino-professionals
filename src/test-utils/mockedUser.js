const mockedUser = {
  id: '9c6cb22e-f284-4ca4-9a88-d64ece7b896c',
  createdAt: '2021-09-02T03:55:52.338Z',
  updatedAt: '2021-09-02T03:55:52.338Z',
  email: 'test@test.com',
  name: 'Testerson da Silva',
  description: null,
  avatarUrl: 'www.avatarurlfake.com',
  phone: '(11) 11111-1112',
  state: 'Minas Gerais',
  city: '√Åguas Formosas',
  hasWhatsapp: false,
};
export default mockedUser;
