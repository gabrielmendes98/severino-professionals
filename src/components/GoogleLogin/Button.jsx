import PropTypes from 'prop-types';
import { O_AUTH_PROVIDERS } from 'commons/constants';
import { GoogleLogin } from './style';

const GoogleLoginButton = ({ handleOAuthLogin, handleOAuthFailure }) => (
  <GoogleLogin
    clientId={process.env.REACT_APP_GOOGLE_CLIENT_ID}
    buttonText="Entrar com Google"
    onSuccess={response => handleOAuthLogin(response, O_AUTH_PROVIDERS.GOOGLE)}
    onFailure={handleOAuthFailure}
    cookiePolicy={'single_host_origin'}
  />
);

GoogleLoginButton.propTypes = {
  handleOAuthLogin: PropTypes.func.isRequired,
  handleOAuthFailure: PropTypes.func.isRequired,
};

export default GoogleLoginButton;
