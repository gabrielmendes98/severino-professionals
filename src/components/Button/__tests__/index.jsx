import { toMatchSnapshot } from 'test-utils';
import Button from '..';

it('should match snapshot', () => {
  toMatchSnapshot(<Button>Test button</Button>);
});

it('should match snapshot with custom color', () => {
  toMatchSnapshot(<Button color="red">Test button</Button>);
});
