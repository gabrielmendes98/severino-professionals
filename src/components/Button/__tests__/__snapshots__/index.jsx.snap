// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Test button
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
